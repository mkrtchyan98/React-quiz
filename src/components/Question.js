import React,{ Component } from 'react';
import Timer from './Timer';

export default class Card extends Component {
  render() {
    const {
      difficulty,
      question,
      duration,
      wrongAnswers,
      correctAnswer,
      checkAnswerFn
    } = this.props;
    const answers = [correctAnswer].concat(wrongAnswers).sort();
    return (
      <article className="card">
        <header>
          <span>{difficulty} | </span>
          <Timer duration={duration} timeoutFn={checkAnswerFn(true, false)} />
        </header>
        <div>
          <p>{question}</p>
        </div>
        <footer>
          {answers.map((answer, i) => {
            return (
              <button key={i} onClick={checkAnswerFn(answer, correctAnswer)}>
                {answer}
              </button>
            );
          })}
        </footer>
      </article>
    );
  }
}