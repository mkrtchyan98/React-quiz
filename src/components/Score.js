import React from 'react';

const Score = ({ refresh, score }) => {
  return (
    <article className="scoreCard">
      <h1>Your Score</h1>
      <p>{score}</p>
      <button onClick={refresh}>Play Again</button>
    </article>
  )
}

export default Score;