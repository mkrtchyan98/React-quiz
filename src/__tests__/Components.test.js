import React from 'react';
import Question from '../components/Question'
import Score from '../components/Score';
import Timer from '../components/Timer';
import { shallow, mount,render } from "enzyme";


function setup() {
  const props = {
    checkAnswerFn: jest.fn(),
    difficulty:'medium',
    question:'wag',
    duration:'10',
    wrongAnswers:['ds','dsd'],
    correctAnswer:['dds']
  }

  const enzymeWrapper = shallow(<Question {...props} />)

  return {
    props,
    enzymeWrapper
  }
}

  describe('Question', () => {
    it('snapshot renders', () => {
      const { enzymeWrapper } = setup()
      expect(enzymeWrapper).toMatchSnapshot()
    })
    it('should render self and subcomponents',() => {
       const { enzymeWrapper } = setup()
      expect(enzymeWrapper.find('article').hasClass('card')).toBe(true)
     expect(enzymeWrapper.find('span').text()).toBe('medium | ')
     expect(enzymeWrapper.find('p').text()).toBe('wag')
    })
  })
  


  describe('Score', () => {
  test("snapshot renders",() => {
     const wrapper = shallow(<Score />);
    expect(wrapper).toMatchSnapshot();
  })
})


   describe('Timer',() => {
  test("snapshot renders",() => {
    const wrapper = shallow(<Timer />);
    expect(wrapper).toMatchSnapshot();
  });
  it("shows right left time",() => {
  const wrapper = shallow(<Timer duration={6}/>);
  expect(wrapper.find('span').text()).toEqual('Time Left: 6')
  })
 })
 
