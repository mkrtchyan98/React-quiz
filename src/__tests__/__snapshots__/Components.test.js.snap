// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question snapshot renders 1`] = `ShallowWrapper {}`;

exports[`Score snapshot renders 1`] = `ShallowWrapper {}`;

exports[`Timer snapshot renders 1`] = `ShallowWrapper {}`;
